File: src/main/java/com/applesauce/controller/DogController.java
package com.applesauce.controller;

import com.applesauce.model.Dog;
import com.applesauce.service.DogService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * Created by JacksonChang408 on 5/19/2016.
 */
@RestController
@RequestMapping("/dog")
public class DogController {

    @Autowired
    private DogService dogService;

    @RequestMapping(value = "/getalldogs", method = RequestMethod.GET)
    public List<Dog> getAllDogs(){
        return dogService.getAllDogs();
    }


    @RequestMapping(value = "/adddog", method = RequestMethod.PUT)
    public Dog addDog(@RequestParam("dogName") String dogName,
                      @RequestParam("dogSex") String dogSex,
                      @RequestParam("dogWeight") int dogWeight,
                      @RequestParam("dogBreed") String dogBreed,
                      @RequestParam("dogSpayed") boolean dogSpayed,
                      @RequestParam("dogNotes") String dogNotes){
        return dogService.addDog(new Dog(dogName,dogSex,dogWeight,dogBreed,dogSpayed,dogNotes));
    }
}

File: src/main/webapp/js/controllers/DogController.js
/**
 * Created by JacksonChang408 on 6/12/2016.
 */
app.controller('DogController', ['$scope', '$http', function($scope, $http){


    $scope.getAllDogs = function() {

        $http({method : 'GET',
            url : '/dog/getalldogs'})
            .success(function(data, status) {
                $scope.dogData = data;
                $scope.status = status;
            })
            .error(function(data, status) {
                alert("Error");
            });
    };
    $scope.addDog = function(dogName,dogSex,dogWeight,dogBreed,dogSpayed,dogNotes){
        $http({method : 'PUT',
            url : '/dog/adddog',
            params : {"dogName": dogName,
                "dogSex": dogSex,
                "dogWeight": dogWeight,
                "dogBreed": dogBreed,
                "dogSpayed": dogSpayed,
                "dogNotes" : dogNotes
            }})
            .success(function(data, status) {
                $scope.dogData = data;
                $scope.status = status;
            })
            .error(function(data, status) {
                alert(status);
            });
    };

}]);


